ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx

```diff
@@ -85,6 +95,25 @@ export const PodsLogsViewer = (props: {applicationName: string; pod: models.Reso
                             }}>
                             {prefs.appDetails.darkMode ? <i className='fa fa-sun' /> : <i className='fa fa-moon' />}
                         </div>
+                        <div style={{display: 'flex', marginLeft: 'auto'}}>
+                            <button
+                                className={`argo-button argo-button--base${filter.inverse ? '' : '-o'}`}
+                                onClick={() => setFilter({...filter, inverse: !filter.inverse})}
+                                style={{marginRight: '10px'}}>
+                                !
+                            </button>
+                            <input
+                                type='text'
+                                placeholder='Filter string'
+                                className='argo-field'
+                                value={filter.literal}
+                                onChange={e => setFilter({...filter, literal: e.target.value})}
+                                style={{padding: 0}}
+                            />
+                            <button onClick={() => loader.reload()} className='argo-button argo-button--base' style={{width: '150px'}}>
```

Comment:
I think it would be more convenient if filter applies automatically when user stop typing . RxJS provides `debounceTime` function so we can do it without coding too much:

```typescript
import {Observable} from 'rxjs';

<button
    className={`argo-button argo-button--base${filter.inverse ? '' : '-o'}`}
    onClick={() => {
        setFilter({...filter, inverse: !filter.inverse});
        loader.reload();
    }}
    style={{marginRight: '10px'}}>
    !
</button>
<input
    ref={(input) => {
        if (input) {
            Observable.fromEvent(input, 'keyup').
               debounceTime(500).
               subscribe(() => loader.reload());
        }
    }}
    type='text'
    placeholder='Filter string'
    className='argo-field'
    value={filter.literal}
    onChange={e => setFilter({...filter, literal: e.target.value})}
    style={{padding: 0}}
/>

```